cmds/jfrogplugin/uploaders/helm/helm.go

```diff
@@ -0,0 +1,248 @@
+package helm
+
+import (
+	"context"
+	"fmt"
+	"io"
+	"net/http"
+	"net/url"
+	"path"
+	"strings"
+	"time"
+
+	"github.com/containerd/containerd/reference"
+
+	"ocm.software/ocm/api/credentials"
+	"ocm.software/ocm/api/credentials/cpi"
+	"ocm.software/ocm/api/credentials/identity/hostpath"
+	"ocm.software/ocm/api/ocm/extensions/artifacttypes"
+	"ocm.software/ocm/api/ocm/plugin/ppi"
+	"ocm.software/ocm/api/utils/runtime"
+)
+
+const (
+	NAME = "JFrogHelm"
+
+	// VERSION of the Uploader TODO Increment once stable
+	VERSION = "v1alpha1"
+
+	// ID_HOSTNAME is the hostname of the artifactory server to upload to
+	ID_HOSTNAME = hostpath.ID_HOSTNAME
+	// ID_PORT is the port of the artifactory server to upload to
+	ID_PORT = hostpath.ID_PORT
+	// ID_REPOSITORY is the repository name in JFrog Artifactory to upload to
+	ID_REPOSITORY = "repository"
+
+	// DEFAULT_TIMEOUT is the default timeout for http requests issued by the uploader.
+	DEFAULT_TIMEOUT = time.Minute
+)
+
+type JFrogHelmUploaderSpec struct {
+	runtime.ObjectVersionedType `json:",inline"`
+
+	// URL is the hostname of the JFrog instance.
+	// Required for correct reference to Artifactory.
+	URL string `json:"url"`
+
+	// Repository is the repository to upload to.
+	// Required for correct reference to Artifactory.
+	Repository string `json:"repository"`
+
+	// ChartName is the desired name of the chart in the repository.
+	// OPTIONAL: If not set, defaulted from the passed Hint.
+	ChartName string `json:"name"`
+	// Version is the desired version of the chart
+	// OPTIONAL: If not set, defaulted from the passed Hint.
+	ChartVersion string `json:"version"`
+
+	// Timeout is the maximum duration the upload of the chart can take
+	// before aborting and failing.
+	// OPTIONAL: If not set, set to the internal DEFAULT_TIMEOUT.
+	Timeout time.Duration `json:"timeout"`
```

Comment:
Yah, but what I'm saying is what if someone wants to have `0` as timeout?

Reply:
gotcha... why would someone want that? Do you think disabling timeouts is relevant? I will change to pointer with that in mind.

Reply:
Unsure. Normally, people who set 0 as a timeout either have a very shitty network connection and expect the operation to _take a long time_ or people who upload large data and have no idea how long it would take and don't want to bother with figuring out a timeout value.

Reply:
adjusted to nil value